pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -345,23 +349,21 @@ impl<T: Config> CurrencySwap<<T as frame_system::Config>::AccountId, Balance> fo
 		match Self::account_currency(who) {
 			CORE_ASSET_ID => Ok(PaymentSwapResult::Native),
 			currency => {
-				if T::AMMPool::exists(AssetPair {
+				let amount = if T::AMMPool::exists(AssetPair {
 					asset_in: currency,
 					asset_out: CORE_ASSET_ID,
 				}) {
-					Self::swap_currency(who, fee)?;
-					Ok(PaymentSwapResult::Swapped)
+					T::AMMPool::get_spot_price_unchecked(currency, CORE_ASSET_ID, fee)
```

Comment:
just note that this price can be wrong or manipulated, it will be safer to use  averaged oracle later and even use fallback price instead if the price from oracle is not based on sufficient liquidity

Reply:
https://github.com/galacticcouncil/Basilisk-node/issues/293

Reply:
👍